internal/cli/config.go

```diff
@@ -0,0 +1,48 @@
+package cli
+
+type Config interface {
+	PublicAPIKey() string
+	PrivateAPIKey() string
+	RefreshToken() string
+	AccessToken() string
+}
+
+type userConfig struct {
+	data userConfigData
+}
```

Comment:
So it starts with the interface.  If we ditch it, then yeah, I'm all for just a flat struct with public fields and all that.  If we keep the interface, then the internal data struct really serves as a buffer to allow interface's method and struct's fields to match in name.

Actually started with a struct on this one, but changed it to an interface when I had the desire to write a test that asserted on error messages when invalid YAML was in play.  The other idea that I couldn't quite nail down (opposed to this `Config` interface) was making the config store accept a `Reader` of some sort when trying to `ReadConfig`, so that the test could simply use `strings.NewReader`.  All ears there if you have an easy explanation for how to get that done...

Reply:
What was the decision on using a yaml file instead of json? 

What would allow us to use the reader here would decouple the ReadConfig and the struct. I know this is a pattern we use in baas, but it's not necessary in this repo.

If we are expecting the yaml structure with everything, then we can do something similar (from where the `ReadConfig` gets called):
```
func (fss *fileSystemConfigStore) Read() (io.Reader, error) {
	if _, err := os.Stat(fss.path); os.IsNotExist(err) {
		return nil, nil
	}
	raw, readErr := ioutil.ReadFile(fss.path)
	if readErr != nil {
		return nil, NewPriviligedErr("failed to read CLI config", readErr)
	}
       return raw, readErr
}

type Reader interface {
    Read() (io.Reader, error) 
}
     
.
.
.   
        raw, err := reader.Read()
       if err != nil {
           return err
      }
	var config userConfig
	if err := yaml.Unmarshal(raw, &config); err != nil {
		return nil, NewErrw("config is invalid yaml", err)
	}
	return &config, nil
}

Reply:
The decision on YAML was carried over from the previous cli, perhaps @UnicodeSnowman has background there but i'm guessing it was sort of picked?

I don't think I'm following this example too well...  (the ellipsis lost me, where are we there?  Who calls `store.Read` and handles the unmarshaling?)  I would still want the "Config Store" responsible for returning an actual config.

When you're around, ping me and let's chat offline here.  I'd like to get moving past that ASAP so others' work can be unblocked